plugins/modules/net_tools/nmcli.py

```diff
@@ -1343,6 +1368,38 @@ def main():
             nmcli.module.fail_json(msg="Please specify a name for the master when type is %s" % nmcli.type)
         if nmcli.ifname is None:
             nmcli.module.fail_json(msg="Please specify an interface name for the connection when type is %s" % nmcli.type)
+    if nmcli.type == 'wifi':
+        if nmcli.wifi:
+            available_options = nmcli.get_available_options('802-11-wireless')
+            unsupported_options = []
+            for name, value in nmcli.wifi.items():
+                if name == 'ssid':
+                    nmcli.module.warn("The option 'wifi.ssid' must only be specified with 'ssid', disregarding")
+                    unsupported_options.append(name)
+                if '802-11-wireless.%s' % name not in available_options:
+                    nmcli.module.warn(
+                        "The option 'wifi.%s' is invalid/unsupported, disregarding. Valid options are ['%s']" % (
+                            name,
+                            "', '".join(available_options).replace('802-11-wireless.', '')
+                        )
+                    )
+                    unsupported_options.append(name)
+            for unsupported_option in unsupported_options:
+                del nmcli.wifi[unsupported_option]
```

Comment:
Why not parameterize with a function so the logic can be reused for `wifi_sec` instead of duplicated?

Reply:
Alright I think that should be better, how does it look now?